fetch/api/response/response-consume.html

```diff
@@ -56,80 +70,134 @@
       });
     }
 
-    function checkBodyFormData(response, expectedBody) {
+    function checkBodyFormDataMultipart(response, expectedBody) {
+      return response.formData().then(function(bodyAsFormData) {
+        assert_true(bodyAsFormData instanceof FormData, "Should receive a FormData");
+        var entryName = "name";
+        var strBody = responseStringToMultipartFormTextData(response, entryName, bodyAsFormData.get(entryName));
+        assert_equals(strBody, expectedBody, "Retrieve and verify response's body");
+        assert_true(response.bodyUsed, "body as formData: bodyUsed turned true");
+     });
+    }
+
+    function checkBodyFormDataUrlencoded(response, expectedBody) {
       return response.formData().then(function(bodyAsFormData) {
         assert_true(bodyAsFormData instanceof FormData, "Should receive a FormData");
+        var entryName = "name";
+        var strBody = entryName + "=" + bodyAsFormData.get(entryName);
+        assert_equals(strBody, expectedBody, "Retrieve and verify response's body");
         assert_true(response.bodyUsed, "body as formData: bodyUsed turned true");
      });
     }
 
-    function checkResponseBody(body, bodyType, bodyConsumingType, checkFunction) {
+    function checkBodyFormDataError(response, expectedBody) {
+      return response.formData().then(
+        function(bodyAsFormData) {
+          assert_unreached("Form data parsing should fail");
+        },
+        function(e) {
+          assert_throws(new TypeError(), function() { throw e });
+          assert_true(response.bodyUsed, "body as formData: bodyUsed turned true");
+        });
+    }
+
+    function checkResponseBody(body, expectedBody, bodyType, bodyConsumingType, checkFunction, responseInit) {
```

Comment:
> How about moving bodyType and bodyConsumeType at the end.
> And maybe combining them in something like "from XX to YY", the response name being
> "Consume response body: from XX to YY"

Done.

> Also we use have two parameters body and responseInit. Might want to group them together and put them at the start. Or directly pass a response if that improves readability.

I'd rather not pass a response directly because that would change initialization errors from test failures to global errors. However, I created helper functions which create response promises so that the two parameters are grouped together and named logically.